File: python_75/buySellStocks.py
class Solution:
  def maxProfit(self, prices):

    # 5 7 1 8 6 0 9

    min = prices[0]
    max_profit = 0

    for cur in prices:
      if (cur < min):
        min = cur
      max_profit = max(max_profit, cur - min)

    return max_profit
  
sol = Solution()
print(sol.maxProfit([5, 7, 1, 8, 6, 0, 9]))

File: leetcode/src/main/java/org/example/redo/BuySellStocks.java
package org.example.redo;

public class BuySellStocks {
    public static void main(String[] args) {
        int[] nums = {0, 2, 6, 7, 1, 8, 9};
        System.out.println(find(nums));
    }

    public static int find(int[] nums) {
        int maxProfit = 0;
        int min = nums[0];

        for (int i=0; i<nums.length; i++) {
            if (nums[i] < min)
                min = nums[i];
            maxProfit = Math.max(maxProfit, nums[i] - min);
        }
        return maxProfit;
    }
}
